src/cmd/policy.go

```diff
@@ -110,32 +203,116 @@ func RegoFuncGetGithubRepo(ctx rego.BuiltinContext, a, b *ast.Term) (*ast.Term,
 	}
 
 	if org == "" {
-		log.Error().Msgf("opslevel.repo.github(\"%s\", \"%s\") failed: Please provide a valid org", org, repo)
-		return nil, nil
+		err := fmt.Errorf("opslevel.repo.github(\"%s\", \"%s\") failed: Please provide a valid org", org, repo)
+		log.Error().Err(err).Msgf("")
+		return nil, err
 	}
 
 	if repo == "" {
-		log.Error().Msgf("opslevel.repo.github(\"%s\", \"%s\") failed: Please provide a valid repo", org, repo)
-		return nil, nil
+		err := fmt.Errorf("opslevel.repo.github(\"%s\", \"%s\") failed: Please provide a valid repo", org, repo)
+		log.Error().Err(err).Msg("")
+		return nil, err
 	}
 
 	githubToken := viper.GetString("github-token")
 	authorizationHeader := fmt.Sprintf("token %s", githubToken)
 	githubAPIUrl := fmt.Sprintf("https://api.github.com/repos/%v/%v", org, repo)
 
+	var result githubResponse
+
 	response, err := getClientRest().R().
 		SetHeader("Accept", "application/vnd.github+json").
 		SetHeader("Authorization", authorizationHeader).
+		SetResult(&result).
 		Get(githubAPIUrl)
-	cobra.CheckErr(err)
+	if err != nil {
+		return nil, err
+	}
 
 	if response.IsError() == true {
-		log.Error().Msgf("error requesting Github repo metadata. CODE: %d: REASON: %s", response.StatusCode(), response)
-		return nil, nil
+		err := fmt.Errorf("error requesting Github repo metadata. CODE: %d: REASON: %s", response.StatusCode(), response)
+		log.Error().Err(err).Msgf("")
+		return nil, err
 	}
 
-	reader := strings.NewReader(response.String())
-	v, err := ast.ValueFromReader(reader)
+	languagesResponse, err := getClientRest().R().
+		SetHeader("Accept", "application/vnd.github+json").
+		SetHeader("Authorization", authorizationHeader).
+		SetResult(&result.Languages).
+		Get(githubAPIUrl + "/languages")
+	if err != nil {
+		return nil, err
+	}
+
+	if languagesResponse.IsError() == true {
+		err := fmt.Errorf("error requesting Github repo languages. CODE: %d: REASON: %s", response.StatusCode(), response)
+		log.Error().Err(err).Msgf("")
+		return nil, err
+	}
+
+	result.Languages = convertToPercentage(result.Languages)
+	repoMetadata := githubResponseToRepoMetadata(result)
+
+	v, err := toASTValue(repoMetadata)
+	if err != nil {
+		return nil, err
+	}
+	return ast.NewTerm(v), nil
+}
+
+func RegoFuncGetGitlabRepo(ctx rego.BuiltinContext, a *ast.Term) (*ast.Term, error) {
+
+	var path string
+	if err := ast.As(a.Value, &path); err != nil {
+		return nil, err
+	}
+
+	if path == "" {
+		err := fmt.Errorf("opslevel.repo.gitlab(\"%s\") failed: Please provide a valid Gitlab project path", path)
+		log.Error().Err(err).Msgf("")
+		return nil, err
+	}
+
+	escapedPath := url.QueryEscape(path)
+	gitlabToken := viper.GetString("gitlab-token")
+	gitlabAPIUrl := fmt.Sprintf("https://gitlab.com/api/v4/projects/%s", escapedPath)
+
+	var result gitlabResponse
+
+	response, err := getClientRest().R().
+		SetHeader("PRIVATE-TOKEN", gitlabToken).
+		SetResult(&result).
+		Get(gitlabAPIUrl)
+	if err != nil {
```

Comment:
Should I be logging the error here similar to line 292?

Reply:
yes probably